File: apps/api/app/schemas/automation_execution.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.schemas.automation_rule import AutomationActionType, AutomationTriggerType
from app.schemas.base import to_camel


class AutomationExecutionRead(BaseModel):
    id: UUID
    workspace_id: UUID
    automation_rule_id: UUID
    trigger_type: AutomationTriggerType
    trigger_entity_id: UUID
    action_type: AutomationActionType
    result_entity_id: UUID | None
    created_at: datetime

    model_config = ConfigDict(
        alias_generator=to_camel,
        from_attributes=True,
        populate_by_name=True,
    )

File: apps/web/types/automation-execution.ts
import type {
  AutomationActionType,
  AutomationTriggerType,
} from "@/types/automation-rule";

export type AutomationExecution = {
  id: string;
  workspaceId: string;
  automationRuleId: string;
  triggerType: AutomationTriggerType;
  triggerEntityId: string;
  actionType: AutomationActionType;
  resultEntityId: string | null;
  createdAt: string;
};

export type AutomationExecutionFilters = {
  workspaceId?: string;
  automationRuleId?: string;
  triggerEntityId?: string;
};
